modules/Users/Changenewpassword.php

```diff
@@ -98,166 +88,126 @@
-		$add_captcha = 1;
-		$captcha_privatekey = $admin->settings['captcha_private_key'];
-		$captcha_publickey = $admin->settings['captcha_public_key'];
-		$captcha_js .="<script type='text/javascript' src='" . getJSPath('cache/include/javascript/sugar_grp1_yui.js') . "'></script><script type='text/javascript' src='" . getJSPath('cache/include/javascript/sugar_grp_yui2.js') . "'></script>
-		<script type='text/javascript' src='http://www.google.com/recaptcha/api/js/recaptcha_ajax.js'></script>
-		<script> //var oldFormAction = document.getElementById('form').action; //save old action
-		function initCaptcha(){
-				Recaptcha.create('$captcha_publickey' ,'captchaImage',{theme:'custom',callback:Recaptcha.focus_response_field});
-				}
-
-		window.onload=initCaptcha;
-
-		var handleFailure=handleSuccess;
-		var handleSuccess = function(o){
-			if(o.responseText!==undefined && o.responseText =='Success'){
-
-				document.getElementById('username_password').value=document.getElementById('new_password').value;
-				document.getElementById('ChangePasswordForm').submit();
-			}
-			else{
-				alert(o.responseText);
-				Recaptcha.reload();
-
-			}
-		}
-		var callback2 =
-		{
-		  success:handleSuccess,
-		  failure: handleFailure
-		};
-		function validateCaptchaAndSubmit(){
-				var form = document.getElementById('form');
-				var url = '&to_pdf=1&module=Home&action=index&entryPoint=Changenewpassword&recaptcha_challenge_field='+Recaptcha.get_challenge()+'&recaptcha_response_field='+ Recaptcha.get_response();
-				YAHOO.util.Connect.asyncRequest('POST','index.php',callback2,url);
-
-		}
-
-	</script>";
-	$Captcha.=$captcha_js;
-	$Captcha.= "<tr>
-					<td scope='row' width='20%'>".$mod_strings['LBL_RECAPTCHA_INSTRUCTION_OPPOSITE'].":</td>
-		            <td width='70%'><input type='text' size='26' id='recaptcha_response_field' value=''></td>
-		        	<th rowsapn='2' class='x-sqs-list' ><div  id='recaptcha_image'></div></th>
-			    </tr>
-
-			    <tr>
-		         	<td colspan='2'>
-		         		<a href='javascript:Recaptcha.reload()'> ".$mod_strings['LBL_RECAPTCHA_NEW_CAPTCHA']."</a>&nbsp;&nbsp;
-		         		<a class='recaptcha_only_if_image' href='javascript:Recaptcha.switch_type('audio')'>".$mod_strings['LBL_RECAPTCHA_SOUND']."</a>
-		         		<a class='recaptcha_only_if_audio' href='javascript:Recaptcha.switch_type('image')'>".$mod_strings['LBL_RECAPTCHA_IMAGE']."</a>
-		         	</td>
-		        </tr>";
+require_once('include/MVC/View/SugarView.php');
+$view = new SugarView();
+$view->init();
+$view->displayHeader();
+
+$sugar_smarty = new Sugar_Smarty();
+
+$admin = new Administration();
+$admin->retrieveSettings('captcha');
+$add_captcha = 0;
+$captcha_privatekey = "";
+$captcha_publickey = "";
+$captcha_js = "";
+$Captcha = "";
+$captchaContent = "";
+if (isset($admin->settings['captcha_on']) && $admin->settings['captcha_on'] == '1' && !empty($admin->settings['captcha_private_key']) && !empty($admin->settings['captcha_public_key'])) {
```

Comment:
this piece of code seems code duplication of https://github.com/salesagility/SuiteCRM/pull/5531/files#diff-e59e8fed2ab8526ad173560fb20cd3b9R172

Reply:
I have added a util file to remedy the issue (include/utils/recaptcha_utils.php).